Router/Router.swift

```diff
@@ -0,0 +1,539 @@
+import Foundation
+import Alamofire
+import ObjectMapper
+
+public class ResponseData<T> {
+    
+    public let model: T?
+    public let statusCode: Int?
+    public let response: NSURLResponse?
+    public let error: NSError?
+    public let rawData: NSData?
+    
+    public init(model: T?, statusCode: Int?, response: NSURLResponse?, error: NSError?, rawData: NSData?) {
+        self.model = model
+        self.statusCode = statusCode
+        self.response = response
+        self.error = error
+        self.rawData = rawData
+    }
+    
+}
+
+/// Block to be executed when a request has completed.
+public typealias RouterCompletion = (data: NSData?, statusCode: Int?, response: NSURLResponse?, error: NSError?) -> ()
+
+/// Protocol to define the base URL and sample data for an enum.
+public protocol RouteTarget {
+    var baseURL: NSURL { get }
+    var endpoint: Route.Endpoint { get }
+    var parameters: Route.Parameters { get }
+}
+
+public protocol MappableRouteTarget {
+    typealias Model: Mappable
+}
+
+public protocol RouteHeaders {
+    var headers: [String: AnyObject] { get }
+}
+
+/// Protocol to define the opaque type returned from a request
+public protocol Cancellable {
```

Comment:
Why not delete this, rename `CancellableToken` to `Cancellable`,make it public and cancelAction private?